com.adilhanney.super_nonogram.json

```diff
@@ -0,0 +1,107 @@
+{
+	"app-id": "com.adilhanney.super_nonogram",
+	"runtime": "org.freedesktop.Platform",
+	"runtime-version": "22.08",
+	"sdk": "org.freedesktop.Sdk",
+	"command": "super_nonogram",
+	"separate-locales": false,
+	"finish-args": [
+		"--share=network",
+		"--share=ipc",
+		"--socket=fallback-x11",
+		"--socket=wayland",
+		"--device=dri",
+		"--socket=pulseaudio",
+		"--filesystem=xdg-documents/nonogram:create"
+	],
+	"modules": [
+		{
+			"name": "super_nonogram",
+			"buildsystem": "simple",
+			"only-arches": [
+				"x86_64"
+			],
+			"build-commands": [
+				"mkdir -p SuperNonogram",
+				"tar -xf SuperNonogram_*.tar.gz -C SuperNonogram",
+
+				"cp -r SuperNonogram /app/",
+				"chmod +x /app/SuperNonogram/super_nonogram",
+				"mkdir -p /app/bin",
+				"mkdir -p /app/lib",
+				"ln -s /app/SuperNonogram/super_nonogram /app/bin/super_nonogram",
+				"ln -s /app/SuperNonogram/lib/* /app/lib/",
+
+				"install -Dm644 com.adilhanney.super_nonogram.metainfo.xml -t /app/share/metainfo/",
+		
+				"install -Dm644 icon-16x16.png /app/share/icons/hicolor/16x16/apps/com.adilhanney.super_nonogram.png",
+				"install -Dm644 icon-24x24.png /app/share/icons/hicolor/24x24/apps/com.adilhanney.super_nonogram.png",
+				"install -Dm644 icon-32x32.png /app/share/icons/hicolor/32x32/apps/com.adilhanney.super_nonogram.png",
+				"install -Dm644 icon-48x48.png /app/share/icons/hicolor/48x48/apps/com.adilhanney.super_nonogram.png",
+				"install -Dm644 icon-64x64.png /app/share/icons/hicolor/64x64/apps/com.adilhanney.super_nonogram.png",
+				"install -Dm644 icon-128x128.png /app/share/icons/hicolor/128x128/apps/com.adilhanney.super_nonogram.png",
+				"install -Dm644 icon-256x256.png /app/share/icons/hicolor/256x256/apps/com.adilhanney.super_nonogram.png",
+				"install -Dm644 icon-512x512.png /app/share/icons/hicolor/512x512/apps/com.adilhanney.super_nonogram.png",
```

Comment:
Unless you're using it internally, 512z512 is enough.

Reply:
In the [requirements](https://docs.flathub.org/docs/for-app-authors/requirements/#application-icons) it says
> Applications must provide application icons in at least 64×64px and 128×128px sizes. Application icons should either be included as scalable SVGs or high-resolution PNGs.

Does this need updating or would I need 64x64, 128x128, and 512x512?

(In https://github.com/flathub/flathub/pull/4420 I've just used a 512x512 icon: we'll see if the build succeeds)

Reply:
> Unless you're using it internally, 512z512 is enough.

That mean any other size was superfluous.